app/ts/utils/utils.ts

```diff
@@ -0,0 +1,17 @@
+export const bigintSecondsToDate = (seconds: bigint) => {
+	if (seconds > Number.MAX_SAFE_INTEGER) throw new Error(`Too big seconds value: ${ seconds }`)
+	if (seconds < 0) throw new Error(`Got negative seconds: ${ seconds }`)
+	return new Date(Number(seconds) * 1000)
+}
+
+export const dateToBigintSeconds = (date: Date) => BigInt(date.getTime()) / 1000n
+
+export function humanReadableDateDelta(secondsDiff: number) {
+	if (secondsDiff <= 0) return '0 seconds'
+	if (secondsDiff > 3600 * 24 * 1.5) return `${ Math.floor((secondsDiff + 1800) / 3600 / 24) } days`
+	if (secondsDiff > 3600 * 1.5) return `${ Math.floor((secondsDiff + 1800) / 3600) } hours`
+	if (secondsDiff > 60 * 1.5) return `${ Math.floor((secondsDiff + 30) / 60) } minutes`
+	return `${ Math.floor(secondsDiff + 0.5) } seconds`
```

Comment:
While not strictly required, I find that if/elseif/else makes it more clear at a glance that each will return.  It also gives type checking to ensure that because if you forget the `return` in one of the conditions the compiler will start telling you that you are returning `undefined` (breaking something somewhere else) rather than returning the wrong value.
```suggestion
	if (secondsDiff <= 0) return '0 seconds'
	else if (secondsDiff > 3600 * 24 * 1.5) return `${ Math.floor((secondsDiff + 1800) / 3600 / 24) } days`
	else if (secondsDiff > 3600 * 1.5) return `${ Math.floor((secondsDiff + 1800) / 3600) } hours`
	else if (secondsDiff > 60 * 1.5) return `${ Math.floor((secondsDiff + 30) / 60) } minutes`
	else return `${ Math.floor(secondsDiff + 0.5) } seconds`
```